src/notebooklm/_sources.py

```
@@ -463,6 +470,29 @@ async def add_file(
+        # the caller's intent.
+        if title is not None and title != filename:
+            try:
+                renamed = await self.rename(notebook_id, source_id, title)
+                # Preserve the placeholder _type_code so callers still know to
+                # poll for the real type once processing finishes.
+                source = Source(
+                    id=renamed.id,
+                    title=renamed.title,
+                    _type_code=source._type_code,
+                )
+            except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)